src/ipc/capnp/common-types.h

```diff
@@ -16,32 +16,16 @@
 
 namespace ipc {
 namespace capnp {
-//! Use SFINAE to define Serializeable<T> trait which is true if type T has a
-//! Serialize(stream) method, false otherwise.
+//! Detect if type has an Serialize method
 template <typename T>
-struct Serializable {
-private:
-    template <typename C>
-    static std::true_type test(decltype(std::declval<C>().Serialize(std::declval<std::nullptr_t&>()))*);
-    template <typename>
-    static std::false_type test(...);
-
-public:
-    static constexpr bool value = decltype(test<T>(nullptr))::value;
+concept Serializable = requires(T t) {
+    decltype(t.Serialize(std::declval<std::nullptr_t&>()))();
```

Comment:
974e041b794072395e41701cd7a71ebdcc362e48: `serialize.h` already defines a `concept Unserializable`, introduced by @maflcko in #29056.

Though the concept is defined is defined differently there, and either version creates a compiler error when used in both places.

Reply:
Agree it can wait for a followup, but if you have to retouch it would be good to add a comment.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30510#discussion_r1766639028

Updated now to use concepts from serialiize.h instead of defining new ones